File: shellheim/src/types/snippet.ts
// Snippet (reusable command) types

export interface Snippet {
  id: string;
  account_id: string;
  name: string;
  content: string;
  description?: string;
  category?: string;
  created_at: string;
  updated_at: string;
}

export interface CreateSnippetRequest {
  name: string;
  content: string;
  description?: string;
  category?: string;
}

export interface UpdateSnippetRequest {
  name?: string;
  content?: string;
  description?: string;
  category?: string;
}

File: shellheim/src-tauri/src/models/snippet.rs
//! Snippet model

use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct Snippet {
    pub id: String,
    pub account_id: String,
    pub name: String,
    pub content: String,
    pub description: Option<String>,
    pub category: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateSnippetRequest {
    pub name: String,
    pub content: String,
    pub description: Option<String>,
    pub category: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct UpdateSnippetRequest {
    pub name: Option<String>,
    pub content: Option<String>,
    pub description: Option<String>,
    pub category: Option<String>,
}
